### PR TITLE
Change the test logic of case netstress_kill_guest

### PR DESCRIPTION
**Pull request by yanglei-rh**
ID:1949032

Signed-off-by: Lei Yang <leiayang@redhat.com>

### CODE REVIEW

**Comment by yanglei-rh** [comment]
@PaulYuuu @kimi1978wy Please help me review it, thanks a lot.

**Comment by yanglei-rh** [comment]
Test Pass:
(1/1) Host_RHEL.m8.u5.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.5.0.x86_64.io-github-autotest-qemu.netstress_kill_guest.load.q35: PASS (679.31 s)
(1/1) Host_RHEL.m8.u5.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.netstress_kill_guest.load.q35: PASS (900.57 s)

**Comment by yanglei-rh** [comment]
Hi @PaulYuuu Could you pleas help me review it? Thanks a lot.

File: generic/tests/netstress_kill_guest.py [hunk 1]
```diff
@@ -71,19 +127,11 @@ def netload_kill_problem(test, session_serial):
                                    bg_stress_test), logging.info)
             stress_thread = None
             wait_time = int(params.get("wait_bg_time", 60))
-            bg_stress_run_flag = params.get("bg_stress_run_flag")
             vm_wait_time = int(params.get("wait_before_kill_vm"))
-            env[bg_stress_run_flag] = False
-            stress_thread = utils_misc.InterruptedThread(
-                utils_test.run_virt_sub_test, (test, params, env),
-                {"sub_type": bg_stress_test})
```
Comment (PaulYuuu) [inline generic/tests/netstress_kill_guest.py:79]: `bg_stress_test = params.get("background_stress_test", netperf_stress')`
In your patch, the background test will always be set to "netperf", so `background_stress_test` will become meaningless in cfg.

**Comment by PaulYuuu** [comment]
BTW, if there are more cases that include **netperf_test** in the future, please transfer the code to the provider module, to avoid much duplicate code and make it easier to review.

File: generic/tests/netstress_kill_guest.py (hunk 1 above)
Comment (yanglei-rh) [reply generic/tests/netstress_kill_guest.py:79]: Thanks for your reminder, I have changed it.

**Comment by yanglei-rh** [comment]
> BTW, if there are more cases that include **netperf_test** in the future, please transfer the code to the provider module, to avoid much duplicate code and make it easier to review.

@PaulYuuu I added netperf test module, please help me review it again. Thanks a lot.

**Comment by yanglei-rh** [comment]
Test Pass:
(1/1) Host_RHEL.m8.u5.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.5.0.x86_64.io-github-autotest-qemu.netstress_kill_guest.load.q35: PASS (741.66 s)
(1/1) Host_RHEL.m8.u5.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.netstress_kill_guest.load.q35: PASS (896.18 s)
(1/1) Host_RHEL.m8.u5.qcow2.virtio_scsi.up.virtio_net.Guest.Win10.x86_64.io-github-autotest-qemu.netstress_kill_guest.load.q35: PASS (855.29 s)
(1/1) Host_RHEL.m8.u5.qcow2.virtio_scsi.up.virtio_net.Guest.Win10.i386.io-github-autotest-qemu.netstress_kill_guest.load.q35: PASS (928.81 s)
(1/1) Host_RHEL.m9.u0.qcow2.virtio_blk.up.virtio_net.Guest.Win10.i386.io-github-autotest-qemu.netstress_kill_guest.load.q35: PASS (756.66 s)
 (1/1) Host_RHEL.m9.u0.qcow2.virtio_blk.up.virtio_net.Guest.Win10.x86_64.io-github-autotest-qemu.netstress_kill_guest.load.q35: PASS (703.05 s)
(2/4) Host_RHEL.m9.u0.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.9.0.0.x86_64.io-github-autotest-qemu.netstress_kill_guest.load.q35: PASS (703.90 s)

**Review by PaulYuuu — APPROVED** [review]

**Review by kimi1978wy — COMMENTED** [review]
It seems that session is not closed

22:19:51 DEBUG| Check process error: Timeout expired while waiting for shell command to complete: "wmic process where name='netperf.exe' list"    (output: '')
22:19:56 DEBUG| Sending command (safe): ps -C netserver
22:19:56 DEBUG| Sending command: killall netserver
22:19:56 DEBUG| Sending command: echo $?
22:19:56 DEBUG| Sending command (safe): ps -C netserver
22:19:56 INFO | Stop netserver successfully
22:19:56 DEBUG| Sending command: rm -rf /var/tmp/netperf-2.7.1 rm -rf /var/tmp/netperf-2.7.1.tar.bz2
22:19:56 DEBUG| Sending command: echo $?
22:19:56 DEBUG| Sending command:  rm -rf c:\\/netperf.exe

**Comment by yanglei-rh** [comment]
> It seems that session is not closed
> 
> 22:19:51 DEBUG| Check process error: Timeout expired while waiting for shell command to complete: "wmic process where name='netperf.exe' list" (output: '')
> 22:19:56 DEBUG| Sending command (safe): ps -C netserver
> 22:19:56 DEBUG| Sending command: killall netserver
> 22:19:56 DEBUG| Sending command: echo $?
> 22:19:56 DEBUG| Sending command (safe): ps -C netserver
> 22:19:56 INFO | Stop netserver successfully
> 22:19:56 DEBUG| Sending command: rm -rf /var/tmp/netperf-2.7.1 rm -rf /var/tmp/netperf-2.7.1.tar.bz2
> 22:19:56 DEBUG| Sending command: echo $?
> 22:19:56 DEBUG| Sending command: rm -rf c:\/netperf.exe

@kimi1978wy Please help me review it again, thanks.

**Review by kimi1978wy — APPROVED** [review]
LGTM
Yu Wang <wyu@redhat.com>
 (1/2) Host_RHEL.m8.u5.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win8.i386.1.io-github-autotest-qemu.netstress_kill_guest.load.q35: PASS (765.22 s)
 (2/2) Host_RHEL.m8.u5.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win8.x86_64.1.io-github-autotest-qemu.netstress_kill_guest.load.q35: PASS (672.89 s)
 (2/2) Host_RHEL.m9.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win2016.x86_64.io-github-autotest-qemu.netstress_kill_guest.load.q35: PASS (712.73 s)

**Pull request merged**